common.go

```diff
@@ -326,12 +350,13 @@ func encoderForType(typ string) (func([]byte) *string, error) {
 	default:
 		return nil, fmt.Errorf(`unsupported decoder %#v, only string, hex and base64 are supported.`, typ)
 	}
-	return func(data []byte) *string {
+	return func(data []byte) (json.RawMessage, error) {
 		if data == nil {
-			return nil
+			return nullJSON, nil
 		}
-		s := enc(data)
-		return &s
+		// marshaling a string cannot fail.
```

Comment:
Is this always true? Does it mean that we can get rid of checking the `err` value of the Marshal function?

Reply:
It is always true, I believe, but I've changed it to be defensive anyway.